### ISSUE TITLE
Compilation fails when disabling debug symbols

### ISSUE THREAD

**Issue by joancafom** [fluent/fluent-bit#3816]
## Bug Report

**Describe the bug**

I was compiling Fluent-Bit from the source by following [the official documentation](https://docs.fluentbit.io/manual/installation/sources/build-and-install). After cloning the source code, and only when the `FLB_DEBUG` flag is set to `Off`, one of the code dependencies fails to compile. If `FLB_DEBUG` is set to `On`, the problem is suddenly resolved and compilation succeeds.

**To Reproduce**

```
$ git clone https://github.com/fluent/fluent-bit
$ cd fluent-bit/build/
$ cmake -DFLB_DEBUG=Off ../
...
-- Check if the system is big endian
-- Searching 16 bit integer
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/build/fluent-bit/build

$ make
...
 33%] Built target mbedcrypto
Scanning dependencies of target mbedx509
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/certs.c.o
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/pkcs11.c.o
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509.c.o
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c: In function 'mbedtls_x509_dn_gets':
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:789:40: error: '%s' directive output may be truncated writing up to 255 bytes into a region of size 1 [-Werror=format-truncation=]
         ret = mbedtls_snprintf( p, n, "%s", s );
                                        ^~   ~
In file included from /tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:47:
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/include/mbedtls/platform.h:76:41: note: 'snprintf' output between 1 and 256 bytes into a destination of size 1
 #define MBEDTLS_PLATFORM_STD_SNPRINTF   snprintf /**< The default \c snprintf function to use.  */
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/include/mbedtls/platform.h:243:28: note: in expansion of macro 'MBEDTLS_PLATFORM_STD_SNPRINTF'
 #define mbedtls_snprintf   MBEDTLS_PLATFORM_STD_SNPRINTF
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:789:15: note: in expansion of macro 'mbedtls_snprintf'
         ret = mbedtls_snprintf( p, n, "%s", s );
               ^~~~~~~~~~~~~~~~
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c: In function 'mbedtls_x509_serial_gets':
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:827:41: error: '....' directive output truncated writing 4 bytes into a region of size 1 [-Werror=format-truncation=]
         ret = mbedtls_snprintf( p, n, "...." );
                                        ~^~~
In file included from /tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:47:
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/include/mbedtls/platform.h:76:41: note: 'snprintf' output 5 bytes into a destination of size 1
 #define MBEDTLS_PLATFORM_STD_SNPRINTF   snprintf /**< The default \c snprintf function to use.  */
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/include/mbedtls/platform.h:243:28: note: in expansion of macro 'MBEDTLS_PLATFORM_STD_SNPRINTF'
 #define mbedtls_snprintf   MBEDTLS_PLATFORM_STD_SNPRINTF
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:827:15: note: in expansion of macro 'mbedtls_snprintf'
         ret = mbedtls_snprintf( p, n, "...." );
               ^~~~~~~~~~~~~~~~
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c: In function 'mbedtls_x509_sig_alg_gets':
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:864:41: error: ' (' directive output truncated writing 2 bytes into a region of size 1 [-Werror=format-truncation=]
         ret = mbedtls_snprintf( p, n, " (%s, MGF1-%s, 0x%02X)",
                                        ~^
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:864:39: note: assuming directive output of 3 bytes
         ret = mbedtls_snprintf( p, n, " (%s, MGF1-%s, 0x%02X)",
                                       ^~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:864:39: note: assuming directive output of 3 bytes
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:864:39: note: using the range [0, 4294967295] for directive argument
In file included from /tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:47:
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/include/mbedtls/platform.h:76:41: note: 'snprintf' output 17 or more bytes (assuming 23) into a destination of size 1
 #define MBEDTLS_PLATFORM_STD_SNPRINTF   snprintf /**< The default \c snprintf function to use.  */
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/include/mbedtls/platform.h:243:28: note: in expansion of macro 'MBEDTLS_PLATFORM_STD_SNPRINTF'
 #define mbedtls_snprintf   MBEDTLS_PLATFORM_STD_SNPRINTF
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/build/fluent-bit/lib/mbedtls-2.26.0/library/x509.c:864:15: note: in expansion of macro 'mbedtls_snprintf'
         ret = mbedtls_snprintf( p, n, " (%s, MGF1-%s, 0x%02X)",
               ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/build.make:89: lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1851: lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

```

**Expected behavior**

That the source is able to compile even when `FLB_DEBUG` is set to `Off`.

**Screenshots**
N/A

**Your Environment**

* Version used: 1.8.2 (master)
* Configuration: N/A
* Environment name and version (e.g. Kubernetes? What version?):
```
$ gcc --version
gcc (Debian 8.3.0-6) 8.3.0

$ cmake --version
cmake version 3.13.4
```
* Server type and version: N/A
* Operating System and version: 
```
$ uname -a
Linux 9d346b3df8e2 4.15.0-128-generic #131~16.04.1-Ubuntu SMP Wed Dec 9 17:33:47 UTC 2020 x86_64 GNU/Linux
```
* Filters and plugins: N/A

**Additional information**

Using `FLB_DEBUG=On`, the compilation succeeds:

```
$ rm -rf /tmp/build/fluent-bit
$ git clone https://github.com/fluent/fluent-bit
$ cd fluent-bit/build/
$ cmake -DFLB_DEBUG=On ../
...
-- Check if the system is big endian
-- Searching 16 bit integer
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/build/fluent-bit/build

$ make
...
[ 33%] Built target mbedcrypto
Scanning dependencies of target mbedx509
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/certs.c.o
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/pkcs11.c.o
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509.c.o
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509_create.c.o
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509_crl.c.o
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509_crt.c.o
[ 34%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509_csr.c.o
[ 35%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509write_crt.c.o
[ 35%] Building C object lib/mbedtls-2.26.0/library/CMakeFiles/mbedx509.dir/x509write_csr.c.o
[ 35%] Linking C static library ../../../library/libmbedx509.a
[ 35%] Built target mbedx509
...
Scanning dependencies of target out_lib
[100%] Building C object examples/out_lib/CMakeFiles/out_lib.dir/out_lib.c.o
[100%] Linking C executable ../../bin/out_lib
[100%] Built target out_lib
```

**Comment by nokute78** [comment]
I think the issue is mbedtls issue https://github.com/ARMmbed/mbedtls/issues/4233
The patch https://github.com/ARMmbed/mbedtls/pull/4237 is merged from v2.27.
Fluent-bit v1.8.2 uses mbedtls v2.26.

Could you check the patch https://github.com/ARMmbed/mbedtls/pull/4237 or waiting updating mbedtls ?

**Comment by joancafom** [comment]
Thanks for the details @nokute78 !

> Could you check the patch ARMmbed/mbedtls#4237 or waiting updating mbedtls ?

Will the next Fluent-bit release include the patched version for `mbedtls`?

**Comment by nokute78** [comment]
Sorry, I don't know when..
I think it will be next major release.

Note: Release version when mbedtls had been updated.
v1.6.0(Oct 09, 2020): mbedtls v2.24.0
v1.8.0(Jul 07, 2021): mbedtls v2.26.0

**Comment by jkschulz** [comment]
looks like they released the [fix](https://github.com/ARMmbed/mbedtls/commit/2065a8d8af27c6cb1e40c9462b5933336dca7434) in v2.27.0 (v3.0.0 is also out)

![image](https://user-images.githubusercontent.com/35477443/128535464-840fe677-547d-411f-b491-906dc0698d0a.png)

**Comment by nokute78** [comment]
Note: https://github.com/fluent/fluent-bit/pull/3949

**Comment by nokute78** [comment]
@joancafom mbedtls v2.27 has been merged to current master.  
At least, I believe it will be released as v1.9.0.
https://github.com/fluent/fluent-bit/commit/3db3f27bc8b74a4e6f12f8593e244298f1562ffa

I confirmed I can build if `FLB_DEBUG=off` on master.

**Comment by FraPazGal** [comment]
Hello @nokute78,

Indeed, the compilation is working using `FLB_DEBUG=off` as of `v1.8.4`. Thanks for the fix!

Regards,
Francisco de Paz

**Comment by nokute78** [comment]
https://fluentbit.io/announcements/v1.8.4/

v1.8.4 has been released and its mbedtls is v2.27.
This issue is fixed by v1.8.4.
